networkx/convert_matrix.py

```diff
@@ -121,51 +121,103 @@ def to_pandas_dataframe(G, nodelist=None, multigraph_weight=sum, weight='weight'
     df = pd.DataFrame(data=M, index = nodelist ,columns = nodelist)
     return df
 
-def from_pandas_dataframe(df,create_using=None):
+def from_pandas_dataframe(df, source, target, edge_attr=None,
+        create_using=None):
     """Return a graph from Pandas DataFrame.
 
-    The Pandas DataFrame is interpreted as an adjacency matrix for the graph.
+    The Pandas DataFrame should contain at least two columns of node names and
+    zero or more columns of node attributes. Each row will be processed as one
+    edge instance.
+
+    Note: This function uses DataFrame.iterrows, which is not guaranteed to
+    retain the data type in the row. This is only a problem if your row is
+    entirely numeric and a mix of ints and floats. In that case, all values
+    will be returned as floats.
 
     Parameters
     ----------
     df : Pandas DataFrame
-      An adjacency matrix representation of a graph
+      An edge list representation of a graph
 
-    create_using : NetworkX graph
-       Use specified graph for result.  The default is Graph()
+    source : str (or int)
+      Column name (or index if no column names) of the source nodes (for the
+      directed case).
 
-    Notes
-    -----
-    If the numpy matrix has a single data type for each matrix entry it
-    will be converted to an appropriate Python data type.
+    target : str (or int)
+      Column name (or index if no column names) of the target nodes (for the
+      directed case).
 
-    If the numpy matrix has a user-specified compound data type the names
-    of the data fields will be used as attribute keys in the resulting
-    NetworkX graph.
+    edge_attr : str (or int), iterable, True
+      A name (str or index if no column names) or list of names that will be
+      used to retrieve attributes from the row that should be added to the
+      graph as edge attributes. If `True` is passed as the key, all of the
+      remaining columns will be added.
+
```

Comment:
I found the parameter descriptions a little confusing. You're **not** using integer arguments to `source`, `target` and `edge_attr` as indeces but rather (as pandas allows) expect the column names to be integers.